tests/e2e/tests/pilot/util/environment.go

```diff
@@ -212,23 +250,34 @@ func (e *Environment) Setup() error {
 	e.config = model.MakeIstioStore(crdclient)
 
 	if e.Config.Namespace == "" {
-		if e.Config.Namespace, err = util.CreateNamespaceWithPrefix(e.KubeClient, "istio-test-app-", e.Config.UseAutomaticInjection); err != nil { // nolint: lll
+		if e.Config.Namespace, err = util.CreateNamespaceWithPrefix(e.KubeClient[0], "istio-test-app-", e.Config.UseAutomaticInjection); err != nil { // nolint: lll
 			return err
 		}
+		// Create the namespace on the remote cluster if needed
+		if e.Config.KubeConfig[1] != "" {
+			if _, err = util.CreateNamedNamespace(e.KubeClient[1], e.Config.Namespace, e.Config.UseAutomaticInjection); err != nil { // nolint: lll
```

Comment:
I see why you need to have a new function like CreateNamedNamespace(). 

Refer to https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L103
Consider adding a function CreateNamespaceWithPrefixOrName() that has the same parameters as CreateNamespaceWithPrefix with additional parameter ns_name. Move the contents from CreateNamespaceWithPrefix(). CreateNamesacpeWithPrefix() will call this new function with ns_name as Nil. Another new function CreateNamesapceWithName() calls this new function with prefix as Nil, and a valid name.